examples/policy/cluster-policy/README.md

```diff
@@ -0,0 +1,218 @@
+    policies.kyverno.io/description: >-
+      This policy prevents creating or updating RoleBindings
+      and ClusterRoleBindings that grant new permissions to any
+      ServiceAccount actively referenced by an Sandbox resource.
+spec:
+  validationFailureAction: Enforce
+  rules:
+  - name: block-sandbox-sa-bindings
+    match:
+      any:
+      - resources:
+          kinds:
+          - RoleBinding
+          - ClusterRoleBinding
+    validate:
+      message: "Binding to a ServiceAccount that is actively in use by an Sandbox is forbidden."
+      foreach:
+      - list: "request.object.subjects"
+        preconditions:
+          all:
+          - key: "{{element.kind}}"
+            operator: Equals
+            value: "ServiceAccount"
+        context:
+        - name: matchingSandboxes
+          apiCall:
+            urlPath: "/apis/karo.io/v1/namespaces/{{element.namespace}}/sandboxes"
+            jmesPath: "items[?spec.serviceAccountRef.name == '{{element.name}}'] | length(@)"
+        pattern:
+          matchingSandboxes: 0
+
+```
+
+Apply the policy:
+```bash
+kubectl apply -f prevent-sandbox-binding-policy.yaml
+```
+
+## Understanding the Policy Fields
+
+* `spec.validationFailureAction`: Enforce
+Tells Kyverno to actively block any API request that violates the rule.
+
+* `rules[].match`: Triggers on creation or update of any RoleBinding or ClusterRoleBinding.
+
+* `rules[].validate.foreach`: Iterates over request.object.subjects, filtering for ServiceAccount types.
+
+* `context.apiCall`: Calls the K8s API to check for Sandbox resources referencing the given ServiceAccount.
+
+* `pattern`: The request is allowed only if the number of matching sandboxes is 0. If it’s 1 or more, the request is blocked.
+
+
+### Step 4: Verify the Policy is Active
+
+Check that the policy was successfully created and is ready:
+
+```bash
+kubectl get clusterpolicy prevent-sandbox-sa-binding
+
+NAME         ADMISSION   BACKGROUND   READY   AGE   MESSAGE
+prevent-sandbox-sa-binding   true        true         True    36s   Ready
+```
+
+Ensure `READY` is `True`.
+
+## 4. Testing and Verification
+
+To confirm the policy is working, simulate an attempt to grant new permissions to a protected `ServiceAccount`.
+
+### A. Set Up the Test Scenario
+
+`service-account.yaml`
+
+```yaml
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: sandbox-sa
+  namespace: sandbox-ns
+```
+
+`sandbox.yaml`
+
+```yaml
+apiVersion: agents.x-k8s.io/v1alpha1
+kind: Sandbox
+metadata:
+  name: sandbox-example
+  namespace: sandbox-ns
+spec:
+  podTemplate:
+    metadata:
+      labels:
+        sandbox: my-sandbox
+      annotations:
+        test: "yes"
+    spec:
+      serviceAccountName: sandbox-sa
+      containers:
+      - name: my-container
+        image: busybox
+        command: ["/bin/sh", "-c", "sleep 3600"]
+```
+
+```bash
+kubectl apply -f service-account.yaml
+kubectl apply -f agent-sandbox.yaml
```

Comment:
just for completeness. If you are not planning to remove the other files and inline them, this file is missing.

Reply:
i will inline